### DISCUSSION TITLE
Started lagging out of nowhere and I don't know why

### DISCUSSION THREAD

**Discussion by trufflesq** [streamlink/streamlink-twitch-gui#816] [General]
### Checklist

- [X] I understand that **Streamlink Twitch GUI** is just a launcher for **Streamlink**
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink-twitch-gui/blob/master/CONTRIBUTING.md)
- [X] [I have checked the list of open and recently closed issues](https://github.com/streamlink/streamlink-twitch-gui/issues?q=is%3Aissue+sort%3Aupdated-desc)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink-twitch-gui/commits/master)

### Streamlink Twitch GUI version

v1.12.0

### Streamlink version

2.2.0

### Operating system, environment and configuration details

Windows 10 Pro

### Description

My Streamlink Twitch GUI streams have started to lag out of nowhere and I don't know why, I currently have these settings (that I used from the last 2-3 months) it was working fine but today the lag continues to happen (I would turn off low latency mode, but I kind of don't want the stream chat to be behind as well). Also tried switching different players from MPC to VLC and restarting my internet router also didn't help

![streamlink-twitch-gui_FXq6kBM7cv](https://user-images.githubusercontent.com/88832934/129190374-b847e821-a378-4aa2-b8f5-92d3f4fd90f5.png)
![streamlink-twitch-gui_YYpAszFUTH](https://user-images.githubusercontent.com/88832934/129190380-4d106320-0bed-4d88-b08b-3cb998a47cd9.png)

### Debug log

```text
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.8.7
[cli][debug] Streamlink: 2.2.0
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(1.1.0)
[cli][debug] Arguments:
[cli][debug]  --loglevel=debug
[cli][debug]  --rtmp-rtmpdump=rtmpdump.exe
[cli][debug]  --rtmpdump=C:\Program Files (x86)\Streamlink\rtmpdump\rtmpdump.exe
[cli][debug]  --ffmpeg-ffmpeg=C:\Program Files (x86)\Streamlink\ffmpeg\ffmpeg.exe
usage: streamlink [OPTIONS] <URL> [STREAM]
```

**Comment by bastimeyer** [comment]
> My Streamlink Twitch GUI streams have started to lag out of nowhere

Streamlink Twitch GUI is not responsible for any streaming content. It is just a graphical user interface that acts as a launcher for Streamlink. If you're having issues with Streamlink, then this is not the right place to ask, as the issue template should have already told you.

Also, what do you mean with "lag"? Is it a network/connection related problem? Is it a video decoding/rendering related issue? Is it a player configuration issue?

Since you've already posted a debug log from Streamlink, you should check its log output when watching a stream, perferrably with the `--loglevel=trace` parameter, which will add timestamps to each log line. If you're having network issues, then this could potentially give you hints about your problem. Other than that, you should take a look at Streamlink's docs for all the available parameters that let you tweak the stream transport values (make sure that you are using the latest Streamlink development version, as there have been some changes since the last release, which is 2.3.0):
https://streamlink.github.io/latest/cli.html#stream-transport-options

Video decoding/rendering issues or player configuration issues can be debugged via Streamlink's `--verbose-player` argument if the player writes anything useful to `stdout`/`stderr`. If not, then you'll have to figure it out on your own and maybe take a look at your player's documentation.

The Streamlink Twitch GUI settings which you've posted should not cause any issues if your connection is fast and stable enough. The HLS live edge value is 3, which is the default, and low latency streaming will set it to 2 if enabled. You haven't set any additional player parameters, so there's nothing that could cause problems here.

However, if you're having network issues, then you'll need to disable low latency streaming and increase the live edge value by a bit. This will only work if you connection is fast enough to recover from download speed drops though. If it's not fast enough, then you'll have to choose a lower stream quality with a lower bitrate.

**Comment by imatschoolrn11** [comment]
turn up your live edge u can prob put it back later also try using pot players that fixed 4 me
